gc/gc-base/src/main/java/org/projectnessie/gc/identify/IdentifyLiveContents.java

```diff
@@ -0,0 +1,431 @@
+      // This commit-ID has already been visited with the same (or maybe an older/smaller)
+      // cut-off-timestamp, can abort.
+      LOGGER.debug(
+          "live-set#{}: Not submitting task to walk {}, it has already already visited using a"
+              + "compatible cut-off timestamp.",
+          addContents.id(),
+          namedReference);
+      return ReferencesWalkResult.singleShortCircuit(0, 0);
+    }
+
+    LOGGER.info(
+        "live-set#{}: Start walking the commit log of {} using {}.",
+        addContents.id(),
+        namedReference,
+        cutoffPolicy);
+
+    int numCommits = 0;
+    long numContents = 0;
+
+    try (Stream<LogResponse.LogEntry> commits = repositoryConnector().commitLog(namedReference)) {
+
+      LogEntryHolder holder = new LogEntryHolder();
+      String lastCommitId = null;
+
+      for (Spliterator<LogResponse.LogEntry> spliterator = commits.spliterator();
+          spliterator.tryAdvance(holder::set); ) {
+
+        numCommits++;
+
+        LogResponse.LogEntry logEntry = holder.logEntry;
+        CommitMeta commitMeta = logEntry.getCommitMeta();
+        Instant commitTime = commitMeta.getCommitTime();
+        String commitHash = commitMeta.getHash();
+        List<Operation> operations = logEntry.getOperations();
+        if (operations == null) {
+          // Shout never happen, but in case it's a commit without any operation, just ignore it.
+          continue;
+        }
+        if (commitTime == null || commitHash == null) {
+          throw new IllegalStateException("Mandatory information is null in log entry " + logEntry);
+        }
+
+        if (visitedDeduplicator().alreadyVisited(cutoffPolicy.timestamp(), commitHash)) {
+          // This commit-ID has already been visited with the same (or maybe an older/smaller)
+          // cut-off-timestamp, can abort.
+          LOGGER.info(
+              "live-set#{}: Finished walking the commit log of {} using {} after {} commits, commit {} has already been "
+                  + "checked using a compatible cut-off timestamp.",
+              addContents.id(),
+              namedReference,
+              cutoffPolicy,
+              numCommits,
+              commitHash);
+          return ReferencesWalkResult.singleShortCircuit(numCommits, numContents);
+        }
+
+        // The HEAD commit is always live, consult cutoff-policy for all other commits
+        if (lastCommitId == null || !cutoffPolicy.isCutoff(commitTime, numCommits)) {
+          // commit is "live"
+          LOGGER.debug(
+              "live-set#{}: Checking commit {} with {} operations via {}.",
+              addContents.id(),
+              commitHash,
+              operations.size(),
+              namedReference);
+          lastCommitId = commitHash;
+          numContents +=
+              addContents.addLiveContent(
+                  operations.stream()
+                      .filter(operation -> operation instanceof Operation.Put)
```

Comment:
It just occurred to me that we are not cleaning the "dropped table" files when a table dropped time is newer than cutoff time /cutoff of commit. Which is not same as "dropped reference" behaviour. 

It may cause confusion to the users. Better not add dropped table's contents to the live-content-references . 

As there was no table level end to end testcase, it was hard to catch.

Reply:
Created https://github.com/projectnessie/nessie/issues/5210 for tracking/discussions.